### PR TITLE
Memory Grouping : Work around for U280 grouping issue and bug fix for multiple groups

### PR DESCRIPTION
**Pull request by rozumx**
Issue:
  The U280 platform's memory types are incorrectly set (e.g., MEM_DDR4, MEM_DRAM, etc.).  Because the underlying grouping sorting algorithm grouped according to type, the result was two separate groups for each type.  A testcase currently not supported (and discovered by xclbinutil's DRC check).

Resolution:
The grouping algorithm has been updated to:
  1. Ignore the memory type
  2. Detect if there are more then 1 group for a given IP / argument pair.  If detected, no groupings will be performed for that IP / argument pair.

CRs:
  1. CR-1072704 - hbm_simple working for 2020.1_daily_latest but failing for 2020.2_daily_latest
  2. CR-1081877 - [u280]: Invalid memory grouping for L3 HBM designs during hw_emu (2020.2_daily_latest)

### CODE REVIEW

**Review by stsoe — APPROVED** [review]

**Comment by chvamshi-xilinx** [comment]
Marking "do not merge". @saifuddin-xilinx  is going to verify couple of grouping example. Will merge the changes once those passes.

**Comment by saifuddin-xilinx** [comment]
Basic tests are passed. Verified both U280 and U50 platform.

**Pull request merged**
